theano/gof/type.py

```diff
@@ -634,7 +634,7 @@ def c_code(self, node, name, inputs, outputs, sub):
         """ % dict(ctype=self.rtype.ctype, out=outputs[0], inp=inputs[0])
 
     def c_code_cache_version(self):
-        return (0,)
+        return (0, self.rtype.version)
```

Comment:
if self.rtype.version is None (the default), do we want to enable the cache? I think so, but I'm not 100% sure. We normally do the opposite, but I think we should still enable it. If you agree, add a comment that tell we want that here.